cpp/src/arrow/compute/kernels/filter-benchmark.cc

```diff
@@ -68,6 +68,24 @@ static void FilterFixedSizeList1Int64(benchmark::State& state) {
   }
 }
 
+static void FilterString(benchmark::State& state) {
+  RegressionArgs args(state);
+
+  const int64_t array_size = args.size / sizeof(int64_t);
```

Comment:
It doesn't seem right to divide by `int64_t` here.

Reply:
@fsaintjacques could you comment?

Reply:
My intent was to make the benchmark share a memory budget (args.size) and let them size the input. The goal is to be able to compare operations.

Reply:
Right. So the problem here is to generate a random string array roughly consistent with the given memory budget.